README.md
# Grafpix

A growing `Flutter Package` consist of a huge collection of rectified **icons** and **essential utilities** that grant you outstanding user experience in no time.

## Features

- Up-to-date huge collection of ***icons*** from multiple authors.
- Impressive growing ***utilities*** to improve your design patterns.
- **`WYSIWYG` preview** module for `Flutter` in `Visual Studio Code` to speed up development by providing you with **`Visual Aid`**.
- Ease to deploy.

### `>Newly Added` PixButtons

A Stunning **`Glass-Like Buttons`** with **fully** customized properties.

![Release](https://grafpix.com/release/1.2.4/showcase.jpg)

![Showcase](https://grafpix.com/release/1.2.4/release.jpg)

### A huge iconset of `+2500` Icons

- 1403 Icons from [FontAwsome 5.12](https://fontawesome.com/).
- 336 Icons from [Stephen Hutchings](https://www.s-ings.com/).
- 467 Icons from [CaptainIcon](https://mariodelvalle.github.io/CaptainIconWeb/).
- 491 Clean pix icons.

[![Grafpix Release 1.2.4](https://grafpix.com/release/1.2.4/youtube2.jpg)](https://www.youtube.com/watch?v=JGzcL0ahyHo)

### PixLoaders

used to simulate *Future* progress indicators.

| LoaderType |                               Output                                |
| ---------- | ------------------------------------------------------------------- |
| Flashing   | ![Flashing](https://grafpix.com/api/pixloaders.php?loader=flashing) |
| Spinner    | ![Spinner](https://grafpix.com/api/pixloaders.php?loader=spinner)   |
| Rocks      | ![Rocks](https://grafpix.com/api/pixloaders.php?loader=rocks)       |

### Installation

In your `pubspec.yaml` under `dependencies` add the following line:

```yaml
dependencies:
  grafpix: ^1.2.5
```

From your `Terminal`, run the following code:

```sh
  flutter pub get
```

### Example

PixIcon and PixLoader example

```dart
import 'package:flutter/material.dart';
import 'package:grafpix/icons.dart';
import 'package:grafpix/pixloaders/pix_loader.dart';
import 'package:grafpix/pixbuttons/radial.dart';

class Example extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.grey[600],
      body: FutureBuilder(
          future: waitForFuture(),
          builder: (context, snapshot) {
            if (snapshot.hasData)
              // display PixButton with PixIcon when Future ends.
              return Container(
                alignment: Alignment.center,
                child: PixButton(
                    radius: 70.0,
                    icon: PixIcon.pix_android, //PixIcon
                    iconSize: 50.0,
                    iconColor: Colors.white,
                    backgroundColor: Colors.blue,
                    shutter: 0.6,
                    twinkles: true,
                    onPress: () {
                      print('Icon Pressed');
                    }),
              );
            else
              // display PixLoader while Future in progress.
              return PixLoader(
                  loaderType: LoaderType.Flashing,
                  faceColor: Colors.white70);
          }),
    );
  }
}

// Simulates a future call with 5 seconds delay.
Future waitForFuture() async {
  await Future.delayed(Duration(seconds: 5));
  return true;
}
```

[![GRAFPIX](https://grafpix.com/api/readme.php?rm=logo)](http://grafpix.com)

#### Author

> Ahmed Mustafa < ahmed.dev2020@gmail.com >
> if you have suggestions, please feel free to contact me any time.
> your feedback is highly appreciated.

### License

MIT


CHANGELOG.md
# [1.2.5] - 2020-2-5

* Fixes `typicon` font issue.
* Updating Readme.me.

## [1.2.4] - 2020-2-4

* `NEW!` Adding `PixButton`.
* Updating Example.
* Updating Readme.me.

## [1.2.2] - 2020-1-17

* `NEW!` Adding `WYSIWYG Previewer Module`.
* Adding `PixLoader` Widget for Future Progress Indicators.
* Updating Example.
* Updating Readme.me.

## [1.1.11] - 2020-1-2

* Fixes FontAwsomeIcon class and icons rendering issues.
* Updating Readme.me.

## [1.1.9] - 2020-1-2

* Added new set of FontAwsome icons (969 icons).
* Formating lib classes.
* Updating Readme.me.
* Total of 2597 icons added.

## [1.1.7] - 2019-12-31

* Added new set of FontAwsome icons (434 icons).
* Updating Readme.me.
* Total of 1628 icons added.

## [1.1.6] - 2019-12-31

* Added Complete set of Typicons (336 icons).
* Updating Readme.me.
* Total of 1194 icons added.

## [1.1.5] - 2019-12-30

* Added Complete set of Grafpix free web pix icons (491 icons).
* Fixing 1 font issue.
* Updating Readme.me.
* Total of 857 icons added.

## [1.1.4] - 2019-12-30

* Added Complete set of CaptainIcon (367 icons).
* Fixing 98 font names by removing the trailing numbers.
* Updating Readme.me.

## [1.1.3] - 2019-12-29

* Fixes Readme.me.
* Fixes PixIcon Class.
* Fixes Readme.me Usage version.

## [1.1.0] - 2019-12-29

* Added 300+ of CaptainIcon.
* Fixes 23 Icon details.


example/lib/main.dart
// Example

import 'package:flutter/material.dart';
import 'package:grafpix/icons.dart';
import 'package:grafpix/pixloaders/pix_loader.dart';
import 'package:grafpix/pixbuttons/radial.dart';

class Example extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.grey[600],
      body: FutureBuilder(
          future: waitForFuture(),
          builder: (context, snapshot) {
            if (snapshot.hasData)
              // display PixButton with PixIcon when Future ends.
              return Container(
                alignment: Alignment.center,
                child: PixButton(
                    radius: 70.0,
                    icon: PixIcon.pix_android, //PixIcon
                    iconSize: 50.0,
                    iconColor: Colors.white,
                    backgroundColor: Colors.blue,
                    shutter: 0.6,
                    twinkles: true,
                    onPress: () {
                      print('Icon Pressed');
                    }),
              );
            else
              // display PixLoader while Future in progress.
              return PixLoader(
                  loaderType: LoaderType.Flashing, faceColor: Colors.white70);
          }),
    );
  }
}

// Simulates a future call with 5 seconds delay.
Future waitForFuture() async {
  await Future.delayed(Duration(seconds: 5));
  return true;
}
